ivre/nmapout.py

```diff
@@ -207,12 +207,26 @@ def displayhosts(recordsgen, out=sys.stdout, **kargs):
     """
     if isinstance(recordsgen, dict):
         recordsgen = [recordsgen]
-    for record in recordsgen:
-        displayhost(record, out=out, **kargs)
-        if os.isatty(out.fileno()):
-            input()
-        else:
-            out.write("\n")
+    try:
+        for record in recordsgen:
+            displayhost(record, out=out, **kargs)
+            if os.isatty(out.fileno()):
+                try:
+                    input()
+                except EOFError:
+                    # Gracefully exit pagination on Ctrl+D
+                    out.write("\n")
+                    return
+                except KeyboardInterrupt:
+                    # Tidy output, then exit with the conventional SIGINT code
+                    out.write("\n")
+                    raise SystemExit(130) from None
```

Comment:
I don't think this is needed since the outer try-except block will catch this exception
```suggestion
```